drivers/serial/Kconfig.renesas_ra

```diff
@@ -8,6 +8,7 @@ config UART_RENESAS_RA
 	depends on DT_HAS_RENESAS_RA_UART_SCI_ENABLED
 	select SERIAL_HAS_DRIVER
 	select SERIAL_SUPPORT_INTERRUPT
+	select PINCTRL
```

Comment:
@nordicjm For some SoCs including RA family, PINCTRL is selected by default in `zephyr/soc/**/Kconfig.defconfig`. Should we also remove it from there?

Reply:
@KhiemNguyenT Please handle this with the rest of the RA boards.
I am approving this PR.